core/transactions/staged_mutation.cxx

```diff
@@ -165,14 +184,21 @@ staged_mutation_queue::commit(attempt_context_impl* ctx)
                 commit_doc(ctx, item);
                 break;
         }
-    }
+        state.notify_unstage_complete();
```

Comment:
If I'm reading commit_doc() right it can throw.   So is try-finally style logic needed?

Also what is going to be your policy on handling errors from commit_doc and remove_doc.  It's crucial that we don't have lingering background ops continuing after we're done with staged_mutation_queue::commit/rollback.

Please see the 2 rules [here](https://couchbase.slack.com/archives/C014WB8U2MQ/p1695209631970039) for more details.